packages/js/src/executors/release-publish/release-publish.impl.ts

```diff
@@ -220,12 +221,54 @@ export default async function runExecutor(
       stdio: ['ignore', 'pipe', 'pipe'],
     });
 
-    const stdoutData = JSON.parse(output.toString());
+    /**
+     * We cannot JSON.parse the output directly because if the user is using lifecycle scripts, npm will mix its publish output with the JSON output all on stdout.
+     * Additionally, we want to capture and show the lifecycle script outputs as beforeJsonData and afterJsonData and print them accordingly below.
+     */
+    const extractJsonData = extractNpmPublishJsonData(output.toString());
+    if (!extractJsonData.jsonData) {
```

Comment:
If we throw an error from the function, we won't need this block.

Reply:
Please see comment